src/heaps.rs

```diff
@@ -46,4 +50,25 @@ impl Heaps {
     pub fn get_mut(&mut self, index: u32) -> Option<&mut Heap> {
         self.heaps.get_mut(index as usize)
     }
+
+    pub fn execute_partial_query(
```

Comment:
I know this can be a bit hard to do but -- can we add some comments here on what we think this function does? It does not matter if it's precise or not, but we should start having an understanding on this and memory queries, feel free to check the original repo if you have to.